api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -135,19 +150,24 @@ def __next__(self) -> ObsAndXSOMABatch:
         stats.n_soma_batches += 1
 
         pytorch_logger.debug(f"Retrieved SOMA batch: {stats}")
-        return ObsAndXSOMABatch(obs=obs_batch, X=X_batch, stats=stats)
+        return _ObsAndXSOMABatch(obs=obs_batch, X=X_batch, stats=stats)
 
 
 class _ObsAndXIterator(Iterator[ObsAndXDatum]):
     """
-    Iterates through a set of obs and related X rows, specified as ``soma_joinid``s. Encapsulates the batch-based data
-    fetching from SOMA objects, providing row-based iteration.
+    Iterates through a set of ``obs`` and corresponding ``X`` rows, where the rows to be returned are specified by
+    the ``obs_tables_iter`` argument. For the specified ``obs` rows, the corresponding ``X`` data is loaded and
+    joined together. It is returned from this iterator as 2-tuples of ``X`` and obs Tensors.
+
+    Internally manages the retrieval of data in SOMA-sized batches, fetching the next batch of SOMA data as needed.
```

Comment:
or maybe it is a reference to the batch size controlled by `soma_buffer_bytes` param?

Reply:
Yes, they are related. The `soma_buffer_bytes` sets both the `soma.init_buffer_bytes` and `py.init_buffer_bytes` config params. The effects how many obs rows are read in per SOMA "batch", as determined by whatever number of rows fits into the returned obs PyArrow table. For better or for worse (likely the latter), the corresponding X data will be fetched using the same init_buffer_bytes setting, so will be necessarily require multiple reads.  But overall, the `soma_buffer_bytes` user setting provides some control over maximum mem usage. A row-based setting might be better for the user to comprehend, but that's beyond a documentation change for this PR. How much detail do you think is useful in the docs?

Reply:
> How much detail do you think is useful in the docs?

given how much this will impact performance and memory footprint, it seems like it needs to be covered at least at a high level.  Important for anyone tuning.

It could be covered in a "performance" notebook/doc, rather than the docstrings.

Reply:
I agree and I like the notebook tuning example suggestion! Since there is clear work to be done on fixing memory usage, and which may result in changes to how memory usage is configured, I'm going to defer any doc updates (and new notebook examples) for now, and will circle back to this, armed with more confidence in the impact of this setting, or possibly new settings.  I've updated the relevant bullet point in the [epic](https://app.zenhub.com/workspaces/single-cell-5e2a191dad828d52cc78b028/issues/gh/chanzuckerberg/single-cell/450) covering this work.  Thanks!